lib/readwise/client.rb

```diff
@@ -21,6 +21,44 @@ def create_highlights(highlights: [])
       raise NotImplementedError
     end
 
+    def get_highlight(highlight_id:)
+      url = 'https://readwise.io/api/v2/highlights/' + highlight_id
+      highlight_uri = URI.parse(url)
+      highlight_req = Net::HTTP::Get.new(highlight_uri)
+      highlight_req['Authorization'] = "Token #{@token}"
+      highlight_res = Net::HTTP.start(highlight_uri.hostname, highlight_uri.port, use_ssl: true) do |http|
+        http.request(highlight_req)
+      end
+      raise Error, 'Export request failed' unless highlight_res.is_a?(Net::HTTPSuccess)
+
+      parsed_highlight = JSON.parse(highlight_res.body)
+
+      Highlight.new(
+        book_id: parsed_highlight['book_id'].to_s,
+        color: parsed_highlight['color'],
+        created_at: parsed_highlight['created_at'],
+        end_location: parsed_highlight['end_location'],
+        external_id: parsed_highlight['external_id'],
+        highlight_id: parsed_highlight['id'].to_s,
+        highlighted_at: parsed_highlight['highlighted_at'],
+        is_discard: parsed_highlight['is_discard'],
+        is_favorite: parsed_highlight['is_favorite'],
+        location: parsed_highlight['location'],
+        location_type: parsed_highlight['location_type'],
+        note: parsed_highlight['note'],
+        readwise_url: parsed_highlight['readwise_url'],
+        tags: parsed_highlight['tags'].map do |tag|
+          Tag.new(
+            tag_id: tag['id'].to_s,
+            name: tag['name']
+          )
+        end,
+        text: parsed_highlight['text'],
+        updated_at: parsed_highlight['updated_at'],
+        url: parsed_highlight['url'],
+      )
+    end
+
```

Comment:
@andjosh Thought I'd push this initial bit of code and open things up for discussion. I see that there are some conventions around hiding the implementation of the api endpoint request and just exposing the returning of the data in the form of our structs. Is there a preference for if we want to split this method up the same way as we do for exports (make the api request private and make the public method return the formatted data)?

Also open to any other suggestions of course 👍

Reply:
I think this is generally fine.

I think there's some refactoring that you could do in this PR or that I'll do subsequently if you don't feel up to it:
- Extract the URL base into a class constant
- Make a private method to transform a JSON/Hash Highlight data into a Highlight struct
- Make a private method for sending the generic HTTP request and handling errors and parsing the response body into JSON/Hash

Reply:
Thanks for starting this!

Reply:
Totally down to refactor, I appreciate the feedback! I'd like to get exposed to much more ruby code outside of work, so whether it's refactoring or introducing new code, I'm all ears.

Reply:
Ok! How would you like me to give some examples?

- Add code suggestions/comments on this PR?
- Push up my own branch with a refactor (separate from adding this functionality) for you to review?
- You take a stab at refactoring based on those comments and I review from there?

Reply:
Let's go with the third option, I've already started on some of the above suggestions already haha